### DISCUSSION TITLE
Misclicks. Pain and misery. (Updated)

### DISCUSSION THREAD

**Discussion by VZozul** [sunflower-land/sunflower-land#2347] [Ideas]
After publication, after thinking a lot, I decided to edit the discussion, the idea about shovels is not very good, there are a lot of nuances, everything is fine in this game, why change it, I will leave it below so as not to interfere, but instead I will describe in more detail one of the options that I really like, and I would be happy if it is implemented. At least I hope for attention, since I tried very hard and spent time on this.

With the release of new mechanics in the game, I'm talking about quests, many problems appeared, not so significant, but why not improve some points. Yes, I made a mistake, I pressed the wrong place, it seems like my problems, but such things are very frustrating, and we don’t need upsets in such a positive game! 

The most ideal solution to remove misclicks is to make the mechanics for any interactions, the same as with trees, stones, iron and gold, only not 3 clicks, but 2 clicks. With an accidental misclick, if you move your attention or click into an empty space, the previous interaction is canceled, just like with trees. There are times when this will get in the way, for example when you need to plant a lot of sunflowers, but giving the choice and the ability to turn on and off the Misclick Protection function solved the problem.
Do you need to spend your time on this? I think yes. Firstly, the game should not bring negative emotions, and secondly, you already have this mechanic, so it will not take so much time, but a lot of people will enable this feature and protect themselves from negative emotions in case of accidental harvest or claiming food, the rest can turn off this feature if they do not need it. And most importantly, now is the era of mobile phones, everything is much smaller there than on a computer, and even when you move the map of the island, you can accidentally hit something, plant something wrong. Over time, you will develop a mobile application in the same way, so this feature will be very useful to everyone.
![DOUBLE](https://user-images.githubusercontent.com/103659539/226197784-ecbd1775-e02c-4276-9395-5ce5da6ee4c5.jpg)
I propose to introduce this mechanics for all types of interactions. Plots, planting and harvesting, fruit trees and chickens. 
For chickens, in principle there are 2 clicks already, open the chicken, the second click is to get the egg. You can just finish it, if you turn your attention to something else, or click in an empty place, the chicken will hide the egg.

Further in the picture it is clear how it will look on the example of Plots.

First, let's turn on Misclick Protection. 
You can turn it off, and you will have everything as it is now.
![Choise](https://user-images.githubusercontent.com/103659539/226197903-40aa6919-7ce7-4cbc-b965-d0fe4d9ac692.jpg)

Now let's plant our beloved Beetroot.
![PLANT](https://user-images.githubusercontent.com/103659539/226198067-58c197db-22aa-41d7-bbda-d677012c0506.jpg)

Time to harvest.
![HARVEST](https://user-images.githubusercontent.com/103659539/226198077-7f8070f6-135e-4515-8f80-32198d8fbfd2.jpg)

### Cooking!
![FOOD](https://user-images.githubusercontent.com/103659539/226198096-2d7ba5e1-a22b-4612-aa0d-3be427b24a39.jpg)
Cooking brings the most pain and disappointment. Everything is very simple here. Firstly, I myself want to choose to close the cooking window for me or leave it. I never understood why the window closes itself when I set it to cook and when it was ready. As I see it, I went into the kitchen, pressed "Cook", the window does not close, if I want, I will close it myself. When it's cooked, "Claim" appears there, the exclamation mark is cool, if I had the cooking window closed, I will see an exclamation mark, go in and click "Claim", then the window does not close, I can click on "Cook" or exit. This will save me from misclicks, and at the same time, if I need to cook a lot of Mashed Potato, then I will stay in this window and press "Cook" then "Claim" and so on as much as I want, and then when I finish it, I will close the window myself. From clicker bots the situation will not worsen, in any case there will be bots. 

![Claim](https://user-images.githubusercontent.com/103659539/226198724-d9dd340a-8bce-4d23-9558-a5fda70a1e61.jpg)

There is another option to make the claim of food in the same way as with the rest, when you turn on the Misclick function, then you will need to click 2 times. But as for me, the first option is better, but the second is the place to be.

First Click
![FIREPIT](https://user-images.githubusercontent.com/103659539/226198197-fda50f37-3429-41e0-950a-832d6cbbe3f6.jpg)

Second Click claims food

Thanks in advance for your attention! And thank you for a great game!

Previous idea. (Not required reading)
What do you need to dig? Shovel! Let us all have a shovel in our tools, without quantity. But for the harvester to work, you just need to select it, as we select an ax or a pickaxe. With a shovel, you can think of a lot of things, import it into new skills, at first you get an ordinary shovel, which adds nothing to you, then if you pump skills for harvest, then it will improve. Different types of shovels. It is possible to launch a series of SFTs unique shovels in the future. For example, for wheat "Scythe", you can get it in skills, or you can run it as SFT. Or for example you are going to run a Magic Bean, run an SFT golden magic shovel +0.25 across all crops there. We will have many different shovels, for different crops. And most importantly, it will protect against accidental misclicks!

Chickens! For chickens, we will need an egg basket!
Again, when we build Hen House, we get a basket for collecting eggs, then if we pump in skills, we get an improved egg basket that will already give +0.1 and so on. 
With axes and pickaxes, you can do the same in skills.
In the near future we will have fishing rods, different ones!
To this, you can increase the choice not from 3 tools or seeds that are displayed, but 5 for example. Or make additional small ones so that they don’t take up space. 
It's even better to make 2 choices at once, the tool you choose is a shovel or an axe, and the second choice is seeds. Thus, having chosen a certain shovel and certain seeds. If the shovel gives me + for planting potatoes, I will choose it and potato seeds. After all, there will be a lot of shovels, different, with different bonuses, one shovel will accelerate the growth of carrots, the second will give more yield for cabbage.

**Comment by DaanMich** [comment]
Totally agree here

Imo a claim button inside the cooking station and a confirmation modal would be way safer and would be less stressful!

**Comment by Chucrutty** [comment]
you guys wanna make bot live easyer?

**Reply by VZozul** [reply]
Nothing will change for bots.  Now they are, and the same number will remain.  The less work for them for 2 clicks?  The program does not care how many clicks to make.

**Comment by cryptokohai** [comment]
No thanks, it's a game, I dont want to have to confirm everything like it's a Westpac transaction.

**Reply by VZozul** [reply]
A very constructive response.
Did you read only one of the options? About confirmation windows, this is one of the worst options.
But you're right about one thing, it's a game! And it should not bring negative emotions, otherwise it will not be played. You cut trees, stones, not with one click, but with 3 clicks. Why then fruit trees, crops and chickens with one click? It can also be implemented there with 2 clicks, with a choice in the settings. If you turn on "Misclick Protection", then everything will be like with trees, only in 2 clicks, if you don't turn it on, then it's the same as now.

**Comment by Life-Is-Great** [comment]
![image](https://user-images.githubusercontent.com/108883042/226103080-b189911d-29fe-41fd-8274-8af527449fdf.png)
If implemented it should say "CLAIM" Where timer is now

**Comment by Maje53** [comment]
We already click a lot while harvesting and planting seeds, and it gets overwhelming after a while.

**Reply by VZozul** [reply]
In this case, I wrote about the ability to turn it on and off.  Let’s say you don’t need this, don’t turn it on) Everyone’s choice, I personally really need it, and I think many others also need it.

**Comment by Joea9** [comment]
I just need to force close MM when I accidentally harvest. And it has to be fast before the autosave work

**Reply by VZozul** [reply]
When was the last time you did this? I used to do this too if I accidentally pressed something, but after the introduction of quests, it no longer works for me. It automatically saves when closed.

**Reply by Joea9** [reply]
Yesterday  when i prepare fo wheat. I think it still works. Maybe you force close when it already saved

**Reply by Joea9** [reply]
I press long back button to force closed mm. How do you do it?

**Reply by VZozul** [reply]
It still works, you're right. I tried it from a mobile, I just abruptly close the program, on an iPhone it is quickly done. But that's not the point, it's not a use case, it doesn't always work, you might miss it. Therefore, for those who wish, some kind of protective mechanism against unwanted actions is needed. Moreover, it is not so hard to implement, I think so, but not sure.
